### ISSUE TITLE
Getting Started with Karpenter

### ISSUE THREAD

**Issue by uttamlive** [aws/karpenter#3922]
I get an ERROR when following every single step that is outlined in, (https://karpenter.sh/v0.27.5/getting-started/getting-started-with-karpenter/) did a copy paste of every single command.

2023-05-22T19:20:33.664Z        INFO    controller.provisioner  found provisionable pod(s)      {"commit": "698f22f-dirty", "pods": 5}
2023-05-22T19:20:33.664Z        INFO    controller.provisioner  computed new machine(s) to fit pod(s)   {"commit": "698f22f-dirty", "machines": 1, "pods": 5}
2023-05-22T19:20:33.664Z        INFO    controller.provisioner  launching machine with 5 pods requesting {"cpu":"5125m","pods":"7"} from types c5n.2xlarge, c6a.24xlarge, r6a.metal, r6in.metal, c6a.16xlarge and 251 other(s)   {"commit": "698f22f-dirty", "provisioner": "default"}
2023-05-22T19:21:14.562Z        ERROR   controller.provisioner  launching machine, creating cloud provider instance, creating instance, with fleet error(s), InvalidParameterValue: Value (KarpenterNodeInstanceProfile-uttambaral-karpenter-demo) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name   {"commit": "698f22f-dirty"}

**Comment by uttamlive** [comment]
I get the error below when running that command, and that is the weird thing that I notice it was not creating that role for some reason. and I thought all the resources should be created if I follow this doc. (https://karpenter.sh/v0.27.5/getting-started/getting-started-with-karpenter/)

`An error occurred (NoSuchEntity) when calling the GetInstanceProfile operation: Instance Profile KarpenterNodeInstanceProfile-uttambaral-karpenter-demo cannot be found.`

**Comment by tzneal** [comment]
That role is created via cloud in this step: https://karpenter.sh/v0.27.5/getting-started/getting-started-with-karpenter/#3-create-a-cluster.

**Comment by uttamlive** [comment]
that is the problem, I am running all the step as it's outlined in that DOC. and it's not getting created and I get that error every time I follow that DOC.

**Comment by tzneal** [comment]
Can you paste the output of the `aws cloudformation deploy` command that you run in the guide?

**Comment by uttamlive** [comment]
@tzneal I successfully resolved the issue by identifying a mistake on my part. I had previously created a cluster using the same documentation. However, when I removed the cluster, I overlooked the instruction to manually remove the associated nodes. one of the policies from the previous build had remained in place, causing the failure. 

The problem arose because the IAM policy that the cloud-formation process attempted to create already existed in my account.

thank you for your help on this,
